### DISCUSSION TITLE
[Kafka topic operator] Support Custom resources in multiple namespaces

### DISCUSSION THREAD

**Discussion by AliGouta** [strimzi/strimzi-kafka-operator#4332] [Q&A]
Hello,

Today, the kafka topic operator supports reconciliation with CRs created in one single namespace ($STRIMZI_NAMESPACE),
With my team we investigated the kafka topic operator code and thought about adding a support to listen and reconcile Custom resources and kafka topics while CRs are created in different namespaces. We have imagined a workflow to implement it, submit our proposal (as a PR) and get your feedback on it but we were wondering if this might be acceptable in Strimzi operator or this is could not be considered and never would be accepted by the project maintainers since it may carry some complexity?

**Comment by scholzj** [comment]
I think you will find some old JIRAs discussing this listing lot of the possible challenges which would need to be dealt with. Among them there are things such as:
* How to deal with `topicX` created in 5 different namespaces with different options
* How to create the topics created directly in Kafka? In which namespace should they be?

So I think you need to be ready to deal with these. Of course all contributions are welcome ... so it would be great if you can help move this forward.
